test/extended/networking/external_gateway.go

```diff
@@ -0,0 +1,46 @@
+package networking
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	exutil "github.com/openshift/origin/test/extended/util"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientset "k8s.io/client-go/kubernetes"
+	frameworkpod "k8s.io/kubernetes/test/e2e/framework/pod"
+)
+
+var _ = Describe("[sig-network] external gateway address", func() {
+	oc := exutil.NewCLI("ns-global")
+
+	InOVNKubernetesContext(func() {
+		f := oc.KubeFramework()
+
+		It("Should validate failure if an external gateway address does not match the address family of the pod", func() {
+			err := createPod(f.ClientSet, f.Namespace.Name, "test-valid-gateway-pod")
+			expectNoError(err)
+			makeNamespaceWithExternalGatewaySet(f, "fd00:10:244:2::6")
```

Comment:
Please provide a comment explaining why this IPv6 address is being used.

Reply:
sure, done.